src/coords.c

```diff
@@ -423,10 +443,16 @@ PyObject* GetCoords(PyArrayObject* arr, int include_z) {
       continue;
     }
     /* get the coordinates */
+    i = cursor;  // initialize the index to use in the return_index loop later
```

Comment:
This took me a little bit to figure out: this is initialized here because `cursor` is then incremented by `get_coordinates` below.  Perhaps expanding on the comment a little bit would make that more clear?

Otherwise it is confusing when you then look at `i <= cursor` in the loop below.

Reply:
It looks like this line is leading to the segfault, for reasons I don't understand.  If I hardcode it to 0, no segfault.